src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1175,7 +1175,17 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             wchar_t buf[512];
             const auto len = FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, nullptr, hr, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), &buf[0], ARRAYSIZE(buf), nullptr);
             const std::wstring_view msg{ &buf[0], len };
-            message = winrt::hstring{ fmt::format(std::wstring_view{ RS_(L"RendererErrorOther") }, hr, msg, parameter) };
+            std::wstring resourceString = RS_(L"RendererErrorOther").c_str();
+            //conditional message construction
+            if (!parameter.empty())
+            {
+                resourceString += L" \"{2}\"";
+                message = winrt::hstring{ fmt::format(std::wstring_view{ resourceString }, hr, msg, parameter) };
+            }
+            else
+            {
+                message = winrt::hstring{ fmt::format(std::wstring_view{ resourceString }, hr, msg) };
+            }

```

Comment:
```suggestion
            std::wstring partialMessage = fmt::format(std::wstring_view{ resourceString }, hr, msg);
            if (!parameter.empty())
            {
                fmt::format_to(std::back_inserter(partialMessage), FMT_COMPILE(LR"( "{2}")"), parameter);
            }
            message = winrt::hstring{ partialMessage };
```

How does this sound? It simplifies the logic (there is only one format call that uses the resource string) and reduces the dependency on the number of format specifiers in the resource (e.g. we don't need to be as careful when we change the resource string)

Reply:
It made sense; learnt something new today! Thank you for the review!